### PR TITLE
[SYCL] Enable clang unroll attribute instead of pragma unroll

### PR DESCRIPTION
**Pull request by vmaksimo**

### CODE REVIEW

File: clang/test/CodeGenSYCL/loop_unroll_host.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// RUN: %clangxx -c -S -emit-llvm %s -o - | FileCheck %s
+// CHECK: br label %for.cond, !llvm.loop ![[COUNT:[0-9]+]]
+
+int main() {
+  // CHECK: ![[COUNT]] = distinct !{![[COUNT]], ![[COUNT_A:[0-9]+]]}
+  // CHECK-NEXT: ![[COUNT_A]] = !{!"llvm.loop.unroll.count", i32 4}
+  [[clang::unroll(4)]]
+  for (int i = 0; i < 100; ++i);
```
Comment (vladimirlaz) [inline clang/test/CodeGenSYCL/loop_unroll_host.cpp]: please add check for do and do-while loop here.

**Comment by vmaksimo** [comment]
Commits **should be squashed** before merge, now waiting for CI checks

**Comment by MrSidims** [comment]
> Commits **should be squashed** before merge, now waiting for CI checks

You can squash the commits without waiting for testing results.

**Comment by vmaksimo** [comment]
> > Commits **should be squashed** before merge, now waiting for CI checks
> 
> You can squash the commits without waiting for testing results.

As I know, it can be done automatically when PR is merged.

**Comment by bader** [comment]
> Enable clang unroll attribute instead of pragma unroll

@vmaksimo, @MrSidims, could you provide more details on motivation for this change?
What is wrong with the pragma unroll?

Also it's stated that "pragma unroll" is supposed to be disabled, but I don't see tests for that. I think it's just unfortunate wording, right?

**Comment by MrSidims** [comment]
> > Enable clang unroll attribute instead of pragma unroll
> 
> @vmaksimo, @MrSidims, could you provide more details on motivation for this change?
> What is wrong with the pragma unroll?

Nothing wrong with the pragma, but making it CXX attributes makes it nicer (also aligns it with other fpga loop attributes, not making users to write mixed-up code, like:
`#pragma unroll 4
[[intelfpga::ivdep]]
for(auto V : Vars) {}`

> Also it's stated that "pragma unroll" is supposed to be disabled, but I don't see tests for that. I think it's just unfortunate wording, right?

Just an unfortunate word indeed.

**Comment by bader** [comment]
> Nothing wrong with the pragma, but making it CXX attributes makes it nicer (also aligns it with other fpga loop attributes, not making users to write mixed-up code, like:
> `#pragma unroll 4 [[intelfpga::ivdep]] for(auto V : Vars) {}`

Please, put this information to the commit message/PR description next time.

**Pull request merged**